app/lib/core/services/database/database_interface.dart

```diff
@@ -10,23 +17,29 @@ abstract class DatabaseInterface {
 
   Future<FirestoreUser> getUser(String userId);
 
-  Future<void> newRecord(AnimalTransportRecord newRecord);
+  Future<void> saveNewAtr(InitialAtr atr);
 
-  Future<dynamic> getRecord(User user);
+  Future<void> updateAtr(InitialAtr atr);
```

Comment:
What confuses me about this is that you are indeed giving the identifier to update, but the `InitialAtr` has no information that updates besides` user ID, atrID, is complete`. 

We could put `AnimalTransportRecord` here, but I think what you're getting at in the PR is that we don't need to hold the temporary ATR and instead just it's parts.

- [ ] I think this should not exist then, if we use the other functions `setField` piecemeal.

But, I think we still need to use `AnimalTransportRecord` for active (in-progress) ATRs. In editing the sections of the ATR we can just hook up its specific `setField` function for updating (your work is very good for that!). 

However, we still allow for editing the whole ATR at once (even if we change the design to split sections into pages). Hence, the collective ATR object still has use. Consider also when the app is brand new, if the shipper has an ATR in progress we still pull the whole record from the database.

Reply:
This is very open to discussion though, talk to me on group chat.

Reply:
Resolved by [other comment](https://github.com/holtzmak/Humane-Transport/pull/172#discussion_r581157964).